Source/WebKit/UIProcess/Inspector/glib/RemoteInspectorHTTPServer.cpp

```diff
@@ -45,8 +45,10 @@ bool RemoteInspectorHTTPServer::start(GRefPtr<GSocketAddress>&& socketAddress, u
 {
     m_server = adoptGRef(soup_server_new("server-header", "WebKitInspectorHTTPServer ", nullptr));
 
+    guint16 port = g_inet_socket_address_get_port(G_INET_SOCKET_ADDRESS(socketAddress.get()));
+
     GUniqueOutPtr<GError> error;
-    if (!soup_server_listen(m_server.get(), socketAddress.get(), static_cast<SoupServerListenOptions>(0), &error.outPtr())) {
+    if (!soup_server_listen_all(m_server.get(), port, static_cast<SoupServerListenOptions>(0), &error.outPtr())) {
```

Comment:
An undesired side effect of this is that the remote inspector server will listen in all the interfaces, and not just in the chosen one given in the listening address, because it is discarding the address. This can be a problem in devices which have multiple interfaces (for example: both wired Ethernet and WiFi), and can result in leaving the remote inspector server exposed unintentionally to public networks.

In the [Soup Server documentation](https://libsoup.org/libsoup-3.0/server-howto.html#adding-listening-sockets) it is made clear that it support listening on both IPv4 and v6 at the same time:

> By default, servers listen for both IPv4 and IPv6 connections; if you don’t want this, use the SOUP_SERVER_LISTEN_IPV4_ONLY or SOUP_SERVER_LISTEN_IPV6_ONLY options.

Using `WEBKIT_INSPECTOR_SERVER=:1234` should be possible to listen in all interfaces, both in IPv4+v6 without needing this change. Using a resolvable host name that resolves to both v4+v6 addresses, e.g. `WEBKIT_INSPECTOR_SERVER=myhostname:1234` should work as well.

I'll take a look to make sure the above works, and try to think a bit to see if we can have some syntax meaning “listen on all addresses of interface `ethX`” because IMO that would be handier.